fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -532,6 +532,10 @@ public CommandProcessingResult updateClient(final Long clientId, final JsonComma
             if (changes.containsKey(ClientApiConstants.isVatRequiredParamName)) {
                 isVatRequired = command.booleanPrimitiveValueOfParameterNamed(ClientApiConstants.isVatRequiredParamName);
                 clientForUpdate.setVatRequired(isVatRequired);
+            } else {
+                //FSAB-13- This else is required because isVatRequired remains false when user changes vat from dropdown.
+                // In this case we need the original checkbox value else the original vat value will become null
+                isVatRequired = clientForUpdate.isVatRequired();
```

Comment:
why not remove this and just check  if (!clientForUpdate.isVatRequired()) {
clientForUpdate.isVatRequired()}

since you are already setting the value in the if statement

Reply:
This is because client can uncheck the vat required checkbox and in this case we need to make the vatvalue null for the client